setup.sh

```diff
@@ -14,29 +14,34 @@ normalize_version() {
 }
 
 latest_version() {
-  jq --arg channel "$1" '.releases | map(select(.channel==$channel)) | first'
+  jq --arg channel "$1" --arg dart_sdk_arch "$ARCHITECTURE" '.releases | map(select(.channel==$channel) | select(.dart_sdk_arch==$dart_sdk_arch)) | first'
 }
 
 wildcard_version() {
-  if [[ $1 == any ]]; then
-    jq --arg version "^$2" '.releases | map(select(.version | test($version))) | first'
+  if [[ $2 == *"v"* ]] || [[ $2 == 1 ]] || [[ $2 == 0 ]]; then  # is legacy version format
+    if [[ $1 == any ]]; then
+    jq --arg version "$2" '.releases | map(select(.version | startswith($version) )) | first'
+    else
+    jq --arg channel "$1" --arg version "$2" '.releases | map(select(.channel==$channel) | select(.version | startswith($version) )) | first'
+    fi
+  elif [[ $1 == any ]]; then
+    jq --arg version "$2" --arg dart_sdk_arch "$ARCHITECTURE" '.releases | map(select(.version | startswith($version)) | select(.dart_sdk_arch==$dart_sdk_arch)) | first'    
   else
-    jq --arg channel "$1" --arg version "^$2" '.releases | map(select(.channel==$channel) | select(.version | test($version))) | first'
+    jq --arg channel "$1" --arg version "$2" --arg dart_sdk_arch "$ARCHITECTURE" '.releases | map(select(.channel==$channel) | select(.version | startswith($version) ) | select(.dart_sdk_arch==$dart_sdk_arch)) | first'
   fi
 }
 
 get_version() {
-  if [[ -z $2 ]]; then
+  if [[ $2 == any ]]; then
     latest_version $1
   else
     wildcard_version $1 $2
   fi
 }
 
 get_version_manifest() {
-  releases_manifest=$(curl --silent --connect-timeout 15 --retry 5 $MANIFEST_URL)
-  version_manifest=$(echo $releases_manifest | get_version $1 $(normalize_version $2))
-
+ releases_manifest=$(curl --silent --connect-timeout 15 --retry 5 $MANIFEST_URL)
```

Comment:
Please revert changes that are not related to the actual issue, git diff will be much easier.